src/main.h

```diff
@@ -550,4 +561,167 @@ static const unsigned int REJECT_ALREADY_KNOWN = 0x101;
 /** Transaction conflicts with a transaction already known */
 static const unsigned int REJECT_CONFLICT = 0x102;
 
+namespace {
```

Comment:
thanks, I'm still trying to understand why in some places they have anonymous namespace and in some places things are declared globally, in some cases with `static` keyword

Reply:
done

Reply:
The problem that is being solves is one of symbol visibility.
Every method is a symbol and you want to avoid duplicates in the linked object (library/application).

The solution is to mark methods are local to the CPP file, which makes them compile-unit local and their symbols will not be seen by other compile-units.

Marking a method in a C++ file as 'static' does this trick.
The same result can be had by putting the methods in an anonymous namespace.
It should be noted that some compilers have bigger problems than others with polluting the global namespace, I believe gcc is pretty good overall. LLVM is even better. In general you won't really find it easy to hit the problem that is being solved.

Personally I prefer putting methods in an anonymous namespace to make clear they are file-local. Preferably all at the top of the file or so, or any other place where they are grouped.

In general; we use namespaces to avoid polluting (dumping everything into) the global namespace. This to avoid name-conflicts.

Reply:
Thank you for this awesome explanation